pyspades/server.py

```diff
@@ -1007,9 +1015,9 @@ def grenade_exploded(self, grenade):
         z = position.z
         if x < 0 or x > 512 or y < 0 or y > 512 or z < 0 or z > 63:
             return
-        x = int(x)
-        y = int(y)
-        z = int(z)
+        x = int(math.floor(x))
```

Comment:
Still wouldn't help - `int(math.floor(float('nan')))` still crashes the same as `int(float('nan'))`.